File: spring-boot-ssl/src/main/java/com/example/HelloService.java
package com.example;

import org.springframework.stereotype.Component;

import com.linecorp.armeria.common.HttpResponse;
import com.linecorp.armeria.server.annotation.Get;

@Component
public class HelloService {
    @Get("/")
    public HttpResponse hello() {
        return HttpResponse.of("Hello, Armeria!");
    }
}

File: armeria-spring-boot-kotlin/src/main/kotlin/com/example/HelloService.kt
package com.example;

import com.linecorp.armeria.server.annotation.Default
import com.linecorp.armeria.server.annotation.Get
import com.linecorp.armeria.server.annotation.Param
import com.linecorp.armeria.server.annotation.ProducesJson
import kotlinx.coroutines.delay
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flowOf
import org.springframework.stereotype.Component

@Component
@ProducesJson
class HelloService {

    @Get("/hello1")
    fun hello1() = Response("Hello!")

    @Get("/hello2")
    suspend fun hello2(@Param @Default("hirakida") name: String): Response {
        delay(1000)
        return Response("Hello, $name!")
    }

    @Get("/hello3")
    fun hello3(@Param @Default("hirakida") name: String): Flow<Response> {
        return flowOf(Response("Hello, $name!"))
    }

    data class Response(val message: String)
}
